components/eventing-controller/controllers/subscription/nats/reconciler_test.go

```diff
@@ -71,76 +70,93 @@ var (
 			ctx := context.Background()
 			natsPort := natsPort + id
 			natsServer, natsURL := startNATS(natsPort)
 			defer reconcilertesting.ShutDownNATSServer(natsServer)
 			cancel = startReconciler(eventTypePrefix, natsURL)
 			defer cancel()
 
-			// create subscriber svc
+			// Create a subscriber service
+			subscriberName := fmt.Sprintf(subscriberNameFormat, id) + "-valid"
 			subscriberSvc := reconcilertesting.NewSubscriberSvc(subscriberName, namespaceName)
 			ensureSubscriberSvcCreated(ctx, subscriberSvc)
 
-			// create subscription
-			subscription := reconcilertesting.NewSubscription(subscriptionName, namespaceName,
+			// Create a subscription
+			subscriptionName := fmt.Sprintf(subscriptionNameFormat, id) + "-valid"
+			subscription := reconcilertesting.NewSubscription(subscriptionName, subscriberSvc.Namespace,
 				reconcilertesting.WithFilter("", eventTypeToSubscribe),
-				reconcilertesting.WithValidSink(namespaceName, subscriberName),
+				reconcilertesting.WithSinkURLFromSvc(subscriberSvc),
 			)
 			ensureSubscriptionCreated(ctx, subscription)
 
-			getSubscription(ctx, subscription).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveCondition(eventingv1alpha1.MakeCondition(
+			Context("testing against an available NATS server", func() {
+				expectedConditions := eventingv1alpha1.MakeCondition(
 					eventingv1alpha1.ConditionSubscriptionActive,
 					eventingv1alpha1.ConditionReasonNATSSubscriptionActive,
-					v1.ConditionTrue, "")),
-			))
+					v1.ConditionTrue, "")
+				getSubscription(ctx, subscription).Should(And(
+					reconcilertesting.HaveSubscriptionName(subscriptionName),
+					reconcilertesting.HaveCondition(expectedConditions),
+				))
+			})
 
-			natsServer.Shutdown()
-			getSubscription(ctx, subscription, timeout, pollingInterval).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveSubscriptionNotReady()),
-			)
+			Context("testing against an unavailable NATS server", func() {
+				natsServer.Shutdown()
+				getSubscription(ctx, subscription, timeout, pollingInterval).Should(And(
+					reconcilertesting.HaveSubscriptionName(subscriptionName),
+					reconcilertesting.HaveSubscriptionNotReady()),
+				)
+			})
 
-			_, _ = startNATS(natsPort)
-			getSubscription(ctx, subscription, timeout, pollingInterval).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveSubscriptionReady()),
-			)
+			Context("testing against a NATS server that is available again", func() {
+				_, _ = startNATS(natsPort)
+				getSubscription(ctx, subscription, timeout, pollingInterval).Should(And(
+					reconcilertesting.HaveSubscriptionName(subscriptionName),
+					reconcilertesting.HaveSubscriptionReady()),
+				)
+			})
 		})
 	})
 }
 
-// testCleanEventTypes tests if the reconciler can create the correct cleanEventTypes from the filters of a Subscription.
```

Comment:
Let's see. I think it might still be beneficial to get a summary of the tests at it's (what is the comment above a function actually called?) description. This might help people who want to add test cases to identify the right place faster and to see what cases are not covered.

On the other hand, this is a lie waiting to happen. If someone adds a step or case without updating the description it will be the opposite of useful.